shm/inc/map.hpp

```diff
@@ -2,9 +2,13 @@
 
 #include <vector>
 
+#include "coordinates.hpp"
 #include "island.hpp"
 
 class Map {
+public:
+    Island* getIsland(const Coordinates& coordinate);
```

Comment:
```suggestion
#include "island.hpp"

class Map {
public:
    Island* getIsland(const Coordinates& coordinates);
```
As `coordinate` is a verb, `coordinates` is a noun :)